docs/book/src/migration/v2vsv3.md

```diff
@@ -2,21 +2,71 @@
 
 This document covers all breaking changes when migrating from v2 to v3.
 
-The details of all changes (breaking or otherwise) can be found in [kubebuilder](https://github.com/kubernetes-sigs/kubebuilder/releases) release notes.
+The details of all changes (breaking or otherwise) can be found in
+[controller-runtime](https://github.com/kubernetes-sigs/controller-runtime/releases),
+[controller-tools](https://github.com/kubernetes-sigs/controller-tools/releases)
+and [kubebuilder](https://github.com/kubernetes-sigs/kubebuilder/releases)
+release notes.
+
+## Common changes
+V3 project uses go modules and the go version 1.15 wheredep is no longer supported.
 
 ## Kubebuilder
 
 - A plugin design was introduced to the project. For more info see the [Extensible CLI and Scaffolding Plugins][plugins-phase1-design-doc]
+
 - The GO supported version was upgraded from 1.13+ to 1.15+
 
-## Project config versions
+- New PROJECT file layout which stores the data used to do the scaffolds with the purpose to support the new features and allow future the CLI provide helpers to it users in the future. 
+
+## New plugin go/v3
+
+<aside class="note warning">
+<h1>Note</h1>
+
+From now on, new projects built with the tool will be using by default the go plugin `go.kubebuilder.io/v3+` which has the support for the new features and APIs.
+</aside>
+
+- Support for `apiextensions.k8s.io/v1` to replace `apiextensions.k8s.io/v1beta1` which was deprecated in Kubernetes version `1.16`.
+
+- Support for `admissionregistration.k8s.io/v1` to replace `admissionregistration.k8s.io/v1beta1` which was deprecated in Kubernetes version `1.16`.
+
+- Support for `cert-manager.io/v1` to replace `cert-manager.io/v1alpha2`. Check [CertManager v1.0 docs]( https://cert-manager.io/docs/installation/upgrading/) for breaking changes.
+
+- The manager flags `--metrics-addr` and `enable-leader-election` were renamed for `--metrics-bind-address` and `--leader-elect` to be more aligned with k8s Kubernetes Components. More info: [#1839][issue-1893] 
+
+- Liveness and Readiness probes are now generate by default using [`healthz.Ping`][healthz-ping] 
+
+- New option to create the projects using ComponentConfig was introduced. For more info see its [enhancement proposal](https://github.com/kubernetes/enhancements/tree/master/keps/sig-cluster-lifecycle/wgs) and the [Component config tutorial][component-config-tutorial]
+
+- Required binaries are now download in the project directory.
+
+- Supports the latest releases version of `controller-runtime` and `controller-tools` 
+
+## Migration Guide v2 to V3
```

Comment:
I'd take these 3 sections and word them a bit more like "Do you want to..."

e.g.

```markdown
## Migrating to KubeBuilder v3...

### ...while staying on the go/v2 plugin

### ...and moving to the `go/v3` plugin
```

The use case for that last heading (migrating just the plugin) isn't quite clear to me

Reply:
I agree. I think that the ideal initial had a list of plugins migrations. 
But IMO make sense we know only cover the first scenarios and then, and we working on with the future needs in the future.